pkg/auth/webapp/session_authflow.go

```diff
@@ -753,6 +753,27 @@ func (s *AuthflowScreenWithFlowResponse) Advance(p string, result *Result) {
 	result.RedirectURI = u.String()
 }
 
+func (s *AuthflowScreenWithFlowResponse) AdvanceWithQuery(p string, result *Result, query string) {
+	q := url.Values{}
+	q.Set(AuthflowQueryKey, s.Screen.StateToken.XStep)
+
+	if len(query) > 0 {
+		m, err := url.ParseQuery(query)
+		if err != nil {
+			panic(err)
+		}
+		for k, v := range m {
+			q[k] = v
+		}
+	}
```

Comment:
This method should take `url.Values` directly because we don't want to deal with query parsing in this method.